src/OrbitGl/ThreadTrack.cpp

```diff
@@ -405,3 +397,100 @@ float ThreadTrack::GetHeaderHeight() const {
   header_height += std::max(0, track_count - 1) * space_between_subtracks;
   return header_height;
 }
+
+void ThreadTrack::OnTimer(const TimerInfo& timer_info) {
+  TimerTrack::OnTimer(timer_info);
+
+  // Get the address of the freshly added TextBox and insert it into the ScopeTree.
+  std::shared_ptr<TimerChain> timer_chain = timers_[timer_info.depth()];
+  CHECK(timer_chain != nullptr);
+  {
+    absl::MutexLock lock(&scope_tree_mutex_);
+    scope_tree_.Insert(timer_chain->Last());
+  }
+}
+
+[[nodiscard]] static inline internal::Rect GetTimerInfoRect(const TimerInfo& timer_info,
+                                                            const internal::DrawData& draw_data,
+                                                            const TimeGraph* time_graph,
+                                                            float world_pos_y, float world_size_y) {
+  double start_us = time_graph->GetUsFromTick(timer_info.start());
```

Comment:
Nit/Personal Preference: `const` for the variables?

Reply:
same as above